code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,138 @@
+import com.adobe.marketing.mobile.messaging.ContentCardSchemaData
+import com.adobe.marketing.mobile.messaging.MessagingConstants
+
+/**
+ * Utility object for creating instances of AEP model classes from Map data structures.
+ * Provides functions to create [AepText], [AepImage], [AepButton], and [AepDismissButton].
+ */
+object ContentCardSchemaDataUtils {
+    /**
+     * Creates an instance of [AepText] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepText].
+     * @return An instance of [AepText].
+     */
+    fun createAepText(map: Map<String, Any>): AepText {
+        return AepText(
+            content = map[MessagingConstants.AepUIKeys.CONTENT] as String
+        )
+    }
+
+    /**
+     * Creates an instance of [AepImage] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepImage].
+     * @return An instance of [AepImage].
+     */
+    fun createAepImage(map: Map<String, Any>): AepImage {
+        return AepImage(
+            url = map[MessagingConstants.AepUIKeys.URL] as? String,
+            darkUrl = map[MessagingConstants.AepUIKeys.DARK_URL] as? String
+        )
+    }
+
+    /**
+     * Creates an instance of [AepButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepButton].
+     * @return An instance of [AepButton].
+     */
+    fun createAepButton(map: Map<String, Any>): AepButton {
+        return AepButton(
+            id = map[MessagingConstants.AepUIKeys.INTERACT_ID] as String,
+            actionUrl = map[MessagingConstants.AepUIKeys.ACTION_URL] as String,
+            text = (map[MessagingConstants.AepUIKeys.TEXT] as Map<String, Any>).let { createAepText(it) }
+        )
+    }
+
+    /**
+     * Creates an instance of [AepDismissButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepDismissButton].
+     * @return An instance of [AepDismissButton].
+     */
+    fun createAepDismissButton(map: Map<String, Any>): AepDismissButton {
+        return AepDismissButton(
+            style = map[MessagingConstants.AepUIKeys.STYLE] as? String ?: "NONE_ICON"
+        )
+    }
+}
+
+/**
+ * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+ *
+ * @param contentCardSchemaData The content card schema data containing the content map.
+ * @return The parsed [AepUITemplate] or null if parsing fails.
+ * @throws IllegalArgumentException If the content map is null or if there is an error during parsing.
+ */
+fun getTemplateModelFromContentCardSchemaData(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+    val contentMap =
+        contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+    try {
+        val id = contentMap[MessagingConstants.AepUIKeys.ID] as String
+        val title =
```

Comment:
`title` is required as per spec. If it is null, empty or not provided can you return a null `AepUITemplate` here